skule_vote/backend/views.py

```diff
@@ -264,3 +271,69 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({"voter_eligible": True})
         except (django.core.signing.BadSignature, KeyError):
             return JsonResponse({"voter_eligible": False})
+
+
+class BallotSubmitView(CreateAPIView):
+    serializer_class = BallotSerializer
+
+    def get_serializer_context(self):
+        # Bypass for swagger schema generator
+        if getattr(self, "swagger_fake_view", False):
+            return super().get_serializer_context() | {
+                "student_number_hash": "debug only"
+            }
+
+        return super().get_serializer_context() | {
+            "student_number_hash": self.request.get_signed_cookie("student_number_hash")
+        }
+
+    def check_permissions(self, request):
+        """
+        Checks that the user is logged in with a valid signed cookie, is eligible to vote in the election, and has not
+        previously voted in this election
+        """
+        try:
+            student_number_hash = self.request.get_signed_cookie("student_number_hash")
+        except (django.core.signing.BadSignature, KeyError):
+            self.permission_denied(
+                request, message="You are not logged in as a valid student."
+            )
+
+        voter = Voter.objects.get(student_number_hash=student_number_hash)
+        try:
+            election_id = request.data["electionId"]
+            election = Election.objects.get(id=election_id)
+        except (KeyError, ValueError, Election.DoesNotExist):
+            self.permission_denied(
+                request,
+                message="You did not provide an election Id or the election does not exist",
+            )
```

Comment:
Error  is caught here instead of in the serializer - decide where you want it caught.